cdap-runtime-ext-dataproc/src/main/java/co/cask/cdap/runtime/spi/provisioner/dataproc/DataProcClient.java

```diff
@@ -209,6 +216,67 @@ public OperationSnapshot createCluster(String name,
       cluster.getClusterName(), convertStatus(cluster.getStatus()), nodes, Collections.emptyMap()));
   }
 
+  // finds ingress firewalls for the configured network that open ports 22 and 443,
+  // returning target tags for those rules
+  private Collection<String> getFirewallTargetTags() throws IOException {
+    // figure out the tag to open port 443
+    FirewallList firewalls = compute.firewalls().list(conf.getProjectId()).execute();
+    List<String> tags = new ArrayList<>();
+    boolean found22 = false;
+    boolean found443 = false;
+    for (Firewall firewall : firewalls.getItems()) {
+      // network is a url like https://www.googleapis.com/compute/v1/projects/<project>/<region>/networks/<name>
+      // we want to get the last section of the path and compare to the configured network name
+      String networkName = Paths.get(URI.create(firewall.getNetwork()).getPath()).getFileName().toString();
```

Comment:
It won't work in Windows, which the path separate is `\`, which is different than the one in URI `/`.

Reply:
ok, i'll just substring it